File: LibraryManagementSystem/Frontend/library-management-system-web/src/app/entities/dtos/library/filter-book-dto.ts
import { AuthorDto } from "./author-dto";
import { CategoryDto } from "./category-dto";
import { PublisherDto } from "./publisher-dto";
import { SubCategoryFilterDto } from "./sub-category-filter-dto";

export interface FilterBookDto {
    authors: AuthorDto[] | null;
    publishers: PublisherDto[] | null;
    categories: CategoryDto[] | null;
    subCategories: SubCategoryFilterDto[] | null;
    publicationYear: number | null;
}

File: LibraryManagementSystem/Backend/LibraryManagementSystem/LibraryManagementSystem.Entities/Dtos/Library/FilterBookDto.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace LibraryManagementSystem.Entities.Dtos.Library
{
    public class FilterBookDto
    {
        // Authors
        [Required]
        [JsonPropertyName("authors")]
        public IEnumerable<AuthorDto>? Authors { get; set; }

        // Categories
        [Required]
        [JsonPropertyName("categories")]
        public IEnumerable<CategoryDto>? Categories { get; set; }

        // Publishers
        [Required]
        [JsonPropertyName("publishers")]
        public IEnumerable<PublisherDto>? Publishers { get; set; }

        // SubCategories
        [Required]
        [JsonPropertyName("subCategories")]
        public IEnumerable<SubCategoryFilterDto>? SubCategories { get; set; }

        // Publication year
        [Required]
        [JsonPropertyName("publicationYear")]
        public short? PublicationYear { get; set; } = null;
    }
}
